protocol/xmrtaker/claim.go

```diff
@@ -117,5 +117,11 @@ func (s *swapState) claimMonero(skB *mcrypto.PrivateSpendKey) (*mcrypto.Address,
 
 	close(s.claimedCh)
 	log.Infof("monero claimed and swept to original account %s", depositAddr)
+	go func() {
+		err = s.Exit()
+		if err != nil {
+			log.Warnf("failed to exit: %v", err)
+		}
+	}()
```

Comment:
Why are we using a background goroutine here? I'm asking, because I also want to understand if the caller of this method is safe if the goroutine completely executes before we return.

Reply:
it's impossible for the goroutine to exit before we return. this code path is executed synchronously on a read from the eventCh, and `Exit()` puts something in the eventCh, which has capacity 1. so if this isn't in a goroutine, the code will deadlock